xray/src/inpaint.rs

```diff
@@ -0,0 +1,296 @@
+                                 x: u32,
+                                 y: u32,
+                                 width: u32,
+                                 height: u32,
+                                 to_x: u32,
+                                 to_y: u32|
+         -> ImageResult<()> {
+            if let Some(neighbor) = image_from(spatial_node_id, output_directory, dir)? {
+                image.copy_from(&neighbor.view(x, y, width, height), to_x, to_y);
+            }
+            Ok(())
+        };
+
+        copy_subimage(Direction::TopLeft, w, h, w, h, 0, 0)?;
+        copy_subimage(Direction::Top, 0, h, 2 * w, h, w, 0)?;
+        copy_subimage(Direction::TopRight, 0, h, w, h, 3 * w, 0)?;
+        copy_subimage(Direction::Right, 0, 0, w, 2 * h, 3 * w, h)?;
+        copy_subimage(Direction::BottomRight, 0, 0, w, h, 3 * w, 3 * h)?;
+        copy_subimage(Direction::Bottom, 0, 0, 2 * w, h, w, 3 * h)?;
+        copy_subimage(Direction::BottomLeft, w, 0, w, h, 0, 3 * h)?;
+        copy_subimage(Direction::Left, w, 0, w, 2 * h, 0, h)?;
+
+        Ok(Some(image))
+    } else {
+        Ok(None)
+    }
+}
+
+fn create_inpaint_image(
+    spatial_node_id: SpatialNodeId,
+    output_directory: &Path,
+    inpaint_distance_px: u8,
+) -> ImageResult<()> {
+    if let Some(mut inpaint_image) = stitched_image(spatial_node_id, output_directory)? {
+        inpaint_image = inpaint(inpaint_image, inpaint_distance_px).expect("Inpaint failed.");
+        let inpaint_image_path = get_inpaint_image_path(spatial_node_id, output_directory);
+        inpaint_image.save(inpaint_image_path)?;
+    }
+    Ok(())
+}
+
+fn interpolate(
+    i: u32,
+    j: u32,
+    this_weight: f32,
+    this: &mut SubImage<&mut RgbaImage>,
+    other: &mut SubImage<&mut RgbaImage>,
+) {
+    let this_pix = this.get_pixel_mut(i, j);
+    let other_pix = other.get_pixel_mut(i, j);
+    // We don't use imageproc::pixelops::interpolate as it doesn't round the result
+    this_pix.apply2(&other_pix, |this_c, other_c| {
+        (this_c as f32 * this_weight + other_c as f32 * (1.0 - this_weight)).round() as u8
+    });
+    *other_pix = *this_pix;
+}
+
+fn interpolate_inpaint_image_with_right(
+    spatial_node_id: SpatialNodeId,
+    output_directory: &Path,
+) -> ImageResult<()> {
+    if let (Some((mut current, current_path)), Some((mut right, right_path))) = (
+        inpaint_image_and_path_from(spatial_node_id, output_directory, None)?,
+        inpaint_image_and_path_from(spatial_node_id, output_directory, Direction::Right)?,
+    ) {
+        let (width, height) = (current.width() / 2, current.height());
+        let mut current_sub = current.sub_image(width, 0, width, height);
+        let mut right_sub = right.sub_image(0, 0, width, height);
+        for i in 0..width {
+            let right_weight = i as f32 / (width - 1) as f32;
+            for j in 0..height {
+                interpolate(i, j, right_weight, &mut right_sub, &mut current_sub);
```

Comment:
Maybe I'm misunderstanding something: but shouldn't we pass `1-right_weight` instead of `right_weight` as the third argument here?